code/datums/job_configs/job_configs.dm

```diff
@@ -0,0 +1,131 @@
+	if(!validate_value(value))
+		return FALSE
+	occupation.total_positions = value
+	return TRUE
+
+/// The number of positions a job can have at the start of the round.
+/datum/job_config_type/starting_positions
+	name = JOB_CONFIG_SPAWN_POSITIONS
+
+/datum/job_config_type/starting_positions/get_compile_time_value(datum/job/occupation)
+	if(is_assistant_job(occupation)) // yeah i know it's not the "compile time" value but this is just to remain parity with the old system groooan
+		return -1
+	return initial(occupation.spawn_positions)
+
+/datum/job_config_type/starting_positions/validate_value(value)
+	if(isnum(value))
+		return TRUE
+	return FALSE
+
+/datum/job_config_type/starting_positions/set_current_value(datum/job/occupation, value)
+	if(!validate_value(value))
+		return FALSE
+	occupation.spawn_positions = value
+	return TRUE
+
+/// The amount of playtime required to join a job (minutes).
+/datum/job_config_type/playtime_requirements
+	name = JOB_CONFIG_PLAYTIME_REQUIREMENTS
+
+/datum/job_config_type/playtime_requirements/get_compile_time_value(datum/job/occupation)
+	return initial(occupation.exp_requirements)
+
+/datum/job_config_type/playtime_requirements/validate_value(value)
+	if(isnum(value))
+		return TRUE
+	return FALSE
+
+/datum/job_config_type/playtime_requirements/set_current_value(datum/job/occupation, value)
+	if(!validate_value(value)) // This will mean that the value was commented out, which means that the server operator didn't want to override the codebase default. So, we skip it.
+		return FALSE
+	occupation.exp_requirements = value
+	return TRUE
+
+/// The amount of time required to have an account to join a job (days).
+/datum/job_config_type/required_account_age
+	name = JOB_CONFIG_REQUIRED_ACCOUNT_AGE
+
+/datum/job_config_type/required_account_age/get_compile_time_value(datum/job/occupation)
+	return initial(occupation.minimal_player_age)
+
+/datum/job_config_type/required_account_age/validate_value(value)
+	if(isnum(value))
+		return TRUE
+	return FALSE
+
+/datum/job_config_type/required_account_age/set_current_value(datum/job/occupation, value)
+	if(!validate_value(value)) // This will mean that the value was commented out, which means that the server operator didn't want to override the codebase default. So, we skip it.
+		return FALSE
+	occupation.minimal_player_age = value
+	return TRUE
+
+/// The required age a character must be to join a job (which is in years).
+/datum/job_config_type/required_character_age
+	name = JOB_CONFIG_REQUIRED_CHARACTER_AGE
+
+/datum/job_config_type/required_character_age/get_compile_time_value(datum/job/occupation)
+	return initial(occupation.required_character_age) || 0 // edge case here, this is typically null by default and returning null causes issues. Returning 0 is a safe default.
+
+/datum/job_config_type/required_character_age/validate_value(value) // here we don't care about the value being out of bounds, just wanna make sure it's a number
```

Comment:
do you want the comment here and not the next line down? having it after the proc def is inconsistent with your pattern in this file